migrations/frontend/1528395746_add_is_primary_to_user_emails.up.sql

```diff
@@ -0,0 +1,16 @@
+BEGIN;
+
+ALTER TABLE user_emails ADD COLUMN is_primary bool DEFAULT false NOT NULL;
+
+-- Use our old logic to set the initial primary address.
+-- From this point we expect it to be set from the UI.
+
+UPDATE user_emails SET is_primary = true
+WHERE (user_id, email) IN
+(SELECT DISTINCT ON (user_id) user_id, email FROM user_emails ORDER BY user_id, (verified_at IS NOT NULL) DESC, created_at ASC, email ASC);
```

Comment:
We were using `QueryRowContext` to run this SQL which returns at most one row, but this SQL itself actually returns all rows. So I think we need to add `LIMIT 1`.
```suggestion
(SELECT DISTINCT ON (user_id) user_id, email FROM user_emails ORDER BY user_id, (verified_at IS NOT NULL) DESC, created_at ASC, email ASC LIMIT 1);
```

Reply:
If that's not the case.. then it's me not know well enough how `DISTINCT` works 😂

Reply:
No, that would mean we'd update only the first email returned by this query!

This looks good to me, but I'd find this version more readable, and I think it's equivalent. Can you check?

```sql
UPDATE user_emails ue SET is_primary = true
FROM (
	SELECT DISTINCT ON (user_id) user_id, email
	FROM user_emails
	WHERE verified_at IS NOT NULL
	ORDER BY user_id ASC, created_at ASC, email ASC
) s
WHERE ue.user_id = s.user_id AND ue.email = s.email
```

Reply:
That won't work since if they only have one address we still want to set it as primary, even if it's not verified.

In your version we filter our non verified addresses. So, I think this is what we want:

```sql
UPDATE user_emails ue SET is_primary = true
FROM (
         SELECT DISTINCT ON (user_id) user_id, email
         FROM user_emails
         ORDER BY user_id ASC, (verified_at IS NOT NULL), created_at ASC, email ASC
     ) s
WHERE ue.user_id = s.user_id AND ue.email = s.email;
```